components/system/components/Tag.js

```diff
@@ -0,0 +1,542 @@
+        {
+          <>
+            {(filteredTags || []).map((tag, index) => (
+              <li
+                key={tag}
+                css={type === "dark" ? STYLES_DROPDOWN_ITEM_DARK : STYLES_DROPDOWN_ITEM}
+                onClick={isActiveTag(index) ? () => handleRemove(tag) : () => handleAdd(tag)}
+                data-item-active={type === "dark" && isActiveTag(index)}
+              >
+                {isActiveTag(index) && (
+                  <div css={STYLES_DROPDOWN_ITEM_ICON}>
+                    <SVG.Check height="16px" />
+                  </div>
+                )}
+                <span>{tag}</span>
+                {isActiveTag(index) && (
+                  <div
+                    css={STYLES_DROPDOWN_ITEM_ICON}
+                    style={{ marginLeft: "auto" }}
+                    className="dismiss"
+                    onClick={(e) => {
+                      e.stopPropagation();
+                      handleRemove(tag);
+                    }}
+                  >
+                    <SVG.Dismiss height="16px" />
+                  </div>
+                )}
+              </li>
+            ))}
+            <li
+              css={type === "dark" ? STYLES_DROPDOWN_ADD_ITEM_DARK : STYLES_DROPDOWN_ADD_ITEM}
+              onClick={() => {
+                if (value) {
+                  handleAdd(value);
+                  setOpen(false);
+                }
+              }}
+            >
+              <SVG.Plus height="16px" />
+              <span style={{ margin: "0 8px" }}>create new tag</span>
+              {value && (
+                <span
+                  css={type === "dark" ? STYLES_TAG_DARK : STYLES_TAG}
+                  className="value"
+                  style={{ margin: 0 }}
+                >
+                  {value}
+                </span>
+              )}
+            </li>
+          </>
+        }
+      </ul>
+    </div>
+  );
+};
+
+export const Tag = ({
+  type,
+  name,
+  tags = [],
+  suggestions = [],
+  style,
+  inputStyles,
+  dropdownStyles,
+  placeholder,
+  onChange,
+  handleClick,
+}) => {
+  const [value, setValue] = React.useState("");
+  const [open, setOpen] = React.useState(false);
+  const [showMore, setShowMore] = React.useState(false);
+
+  const numToDisplay = showMore ? tags.length : 15;
+
+  const _handleRemove = (tag) => {
+    const newTags = [...tags];
+    const tagIndex = tags.indexOf(tag);
+
+    newTags.splice(tagIndex, 1);
+
+    if (onChange) {
+      onChange({ target: { name: "tags", value: newTags } });
+    }
+  };
+
+  const _handleAdd = (value) => {
+    if ((tags || []).find((tag) => tag.toLowerCase() === value.toLowerCase().trim())) {
+      return;
+    }
+
+    if (onChange) {
+      onChange({ target: { name: "tags", value: [...tags, value.trim()] } });
+
+      setValue("");
+    }
+  };
+
+  const _handleChange = (e) => setValue(e.target.value.toLowerCase());
+
+  const _handleKeyPress = (e) => {
+    let regex = /[a-z0-9\s-]/i;
```

Comment:
We decided to not have hyphens in tags, so mind changing this to `[a-z0-9\s]` instead of `[a-z0-9\s-]`?